File: backend/src/app/schemas/index_rebuilds.py
"""Index rebuild job schemas."""

from __future__ import annotations

import uuid
from datetime import datetime
from enum import Enum

from pydantic import BaseModel, ConfigDict


class IndexRebuildStatus(str, Enum):
    QUEUED = "queued"
    RUNNING = "running"
    SUCCEEDED = "succeeded"
    FAILED = "failed"
    CANCELED = "canceled"


class IndexRebuildJobRead(BaseModel):
    """Index rebuild job response."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    kb_id: uuid.UUID
    status: IndexRebuildStatus
    error_message: str | None = None
    stats: dict | None = None
    created_at: datetime
    started_at: datetime | None = None
    finished_at: datetime | None = None

File: frontend/src/services/indexRebuilds.ts
/**
 * 索引重建任务 API 封装
 */
import { apiFetch, apiV1Path } from './http';

export type IndexRebuildStatus = 'queued' | 'running' | 'succeeded' | 'failed' | 'canceled';

export interface IndexRebuildJob {
  id: string;
  kb_id: string;
  status: IndexRebuildStatus;
  error_message: string | null;
  stats: Record<string, unknown> | null;
  created_at: string;
  started_at: string | null;
  finished_at: string | null;
}

/**
 * 获取索引重建任务状态
 */
export async function getIndexRebuildJob(jobId: string): Promise<IndexRebuildJob> {
  return apiFetch<IndexRebuildJob>(apiV1Path(`/index-rebuilds/${jobId}`));
}
